reactive/kotlinx-coroutines-reactor/test/MonoAwaitCancellationStressTest.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2016-2022 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.reactor
+
+import kotlinx.coroutines.*
+import org.junit.*
+import org.reactivestreams.*
+import reactor.core.*
+import reactor.core.publisher.*
+import java.util.concurrent.locks.*
+
+class MonoAwaitCancellationStressTest {
+    private val iterations = 10_000 * stressTestMultiplier
+
+    @Test
+    fun testAwaitCancellationOrder() = runBlocking {
```

Comment:
This test consistently deadlocks on the very first iteration on my machine

Reply:
For me, the behavior here is *extremely* odd. When I run the test, I get dozens of times in a row when *all* iterations pass, and then dozens of times in a row when, like for you, the first iteration hangs.